gapic/templates/%namespace/%name_%version/%sub/services/%service/client.py.j2

```diff
@@ -170,24 +164,27 @@ class {{ service.client_name }}(metaclass={{ service.client_name }}Meta):
                 raise ValueError('When providing a transport instance, '
                                  'provide its credentials directly.')
             self._transport = transport
-        elif transport is not None or (
-            client_options.api_endpoint == self.DEFAULT_ENDPOINT
+        elif client_options is None or (
+            client_options.api_endpoint == None
             and client_options.client_cert_source is None
         ):
-            # Don't trigger mTLS.
+            # Don't trigger mTLS if we get an empty ClientOptions.
             Transport = type(self).get_transport_class(transport)
             self._transport = Transport(
-                credentials=credentials, host=client_options.api_endpoint
+                credentials=credentials, host=self.DEFAULT_ENDPOINT
             )
         else:
-            # Trigger mTLS. If the user overrides endpoint, use it as the mTLS
-            # endpoint, otherwise use the default mTLS endpoint.
-            option_endpoint = client_options.api_endpoint
-            api_mtls_endpoint = self.DEFAULT_MTLS_ENDPOINT if option_endpoint == self.DEFAULT_ENDPOINT else option_endpoint
+            # We have a non-empty ClientOptions, trigger mTLS. If the user
+            # doesn't provide endpoint, use the default mTLS endpoint.
+            if client_options.api_endpoint:
+                api_endpoint = api_mtls_endpoint = client_options.api_endpoint
+            else:
+                api_endpoint = self.DEFAULT_ENDPOINT
+                api_mtls_endpoint = self.DEFAULT_MTLS_ENDPOINT
 
             self._transport = {{ service.name }}GrpcTransport(
```

Comment:
This may be addressed in other documents somewhere, but does mTLS require gRPC, or could it be used over http?

Reply:
it is for http and grpc. Will need to add http mTLS support when microgenerator supports http.